### DISCUSSION TITLE
How to compile with std C++ 20 in Visual Studio Pro 2022? (default member initializers for bit-fields requires at least '/std:c++20')

### DISCUSSION THREAD

**Discussion by colorindarkness** [Sixze/ALS-Refactored#408] [Q&A] [answered]
With the latest update I'm having trouble compiling and running into these errors with the newly converted booleans:
`default member initializers for bit-fields requires at least '/std:c++20'`
It's apparent I'm not using the correct compiler, but I cannot find anywhere in the Project Settings to change this. 
![image](https://github.com/Sixze/ALS-Refactored/assets/9194221/da37404a-9767-491d-9460-5e4d85ce4c89)

**Comment by ameaninglessname** [comment]
It's in the `NMake` section

**Reply by colorindarkness** [reply]
I looked there as well. In Additional Options I've tried both "c++20" and "c++latest. "Am I missing something or need to install something? Windows 10 Pro
![image](https://github.com/Sixze/ALS-Refactored/assets/9194221/08d09dad-2e49-456d-a2fd-464f2795a23c)
![image](https://github.com/Sixze/ALS-Refactored/assets/9194221/67ea6e09-f717-4056-8ab2-3da1a2b690a3)
![image](https://github.com/Sixze/ALS-Refactored/assets/9194221/2afaac31-7383-43e1-9c80-4cbd3ae42edc)

**Reply by colorindarkness** [reply, marked as answer]
This was fixed by adding the following to my projects `*.Target.cs` file:

`CppStandard = CppStandardVersion.Cpp20;`

then I Cleaned and performed a Rebuild as opposed to just Building
